cmd/binlog-collector/collector/collector.go

```diff
@@ -0,0 +1,184 @@
+package collector
+
+import (
+	"bytes"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/percona/percona-xtradb-cluster-operator/cmd/binlog-collector/db"
+	"github.com/percona/percona-xtradb-cluster-operator/cmd/binlog-collector/storage"
+)
+
+type Controller struct {
+	db              db.DB
+	storage         storage.Client
+	lastSet         string
+	pxcServiceName  string
+	pxcUser         string
+	pxcPass         string
+	lastSetFileName string // name for object where the last binlog set will stored
+}
+
+type Config struct {
+	PXCServiceName string
+	PXCUser        string
+	PXCPass        string
+	S3Endpoint     string
+	S3accessKeyID  string
+	S3accessKey    string
+	S3bucketName   string
+}
+
+func New(c Config) (*Controller, error) {
+	s3, err := storage.NewS3(c.S3Endpoint, c.S3accessKeyID, c.S3accessKey, c.S3bucketName, true)
+	if err != nil {
+		return nil, errors.Wrap(err, "new storage manager")
+	}
+	lastSetFileName := "last-set.txt"
+	// get last binlog set stored on S3
+	lastSet, err := s3.GetObject(lastSetFileName)
+	if err != nil && !strings.Contains(err.Error(), "does not exist") {
```

Comment:
there are too many errors with `does not exist` string, like
```
The access key ID you provided does not exist in our records.
The specified bucket does not exist.
```
I think we should check error explicitly, like
```suggestion
	if err != nil && err.code != "NoSuchKey" {
```